applications/StructuralMechanicsApplication/custom_elements/base_shell_element.cpp

```diff
@@ -505,6 +505,25 @@ void BaseShellElement<TCoordinateTransformation>::CalculateOnIntegrationPoints(
     }
 }
 
+template <class TCoordinateTransformation>
+void BaseShellElement<TCoordinateTransformation>::CalculateOnIntegrationPoints(
+    const Variable<ConstitutiveLaw::Pointer>& rVariable,
+    std::vector<ConstitutiveLaw::Pointer>& rValues,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    if (rVariable == CONSTITUTIVE_LAW) {
+        rValues.clear();
+        for (auto p_sec : mSections) {
+            auto vec_integration_points = p_sec->GetConstitutiveLawsVector(GetProperties());
+            rValues.reserve(rValues.size() + vec_integration_points.size());
```

Comment:
I would recommend to resize and the assign, instead of reserve + push_back

Reply:
Why?

Reply:
ah I remembered why I did it in this way, otherwise I would have to also store some indices
Since it is no difference performance wise I would leave it